File: 1 - INICIANTE/python/1040.py
notas = str(input()).split(' ')
n1, n2, n3, n4 = float(notas[0]), float(notas[1]), float(notas[2]), float(notas[3])
media = (n1*2+n2*3+n3*4+n4*1)/10

if media<5.0:
    print('Media: %.1f\n' 'Aluno reprovado.' %media)
elif media>=7.0:
    print('Media: %.1f\n' 'Aluno aprovado.' %media)
else:
    print('Media: %.1f\n' 'Aluno em exame.' %media)
    exame = float(input())
    media_2 = (media+exame)/2
    if media_2>=5.0:
        print('Nota do exame: {}\n' 'Aluno aprovado.\n' 'Media final: {}'.format(exame, media_2))
    else:
        print('Nota do exame: {}\n' 'Aluno reprovado.\n' 'Media final: {}'.format(exame, media_2))

File: 1 - INICIANTE/C++17/1040.cpp
#include <iostream>
#include <iomanip>
#include <cmath>
using namespace std;

int main() {
  double n1,n2,n3,n4,n5,x,y;
  cin >> n1 >> n2 >> n3 >> n4;
  x = ((n1*2)+(n2*3)+(n3*4)+(n4*1))/10;
  cout << fixed << setprecision(1) << "Media: " << x << "\n";
   if (x<5) {
     cout << "Aluno reprovado." << "\n";
   }
   if (x>=7) {
     cout << "Aluno aprovado." << "\n";
   }
   else if ((x<7)&&(x>=5)) {
     cout << "Aluno em exame." << "\n";
     cin >> n5;
     y = (n5+x)/2;
    cout << fixed << setprecision(1) << "Nota do exame: " << n5 << "\n";
     
     if (y>=5) {
       cout << fixed << setprecision(1) << "Aluno aprovado." << "\n" << "Media final: " << y << "\n";
     }
     else {
       cout << "Aluno reprovado." << "\n";
     }
   }
   return 0;
}